CODING_GUIDELINES.md

```diff
@@ -25,6 +25,39 @@ TypeSet with more than one element blocks AND where vCD API returns them without
 *Note*. The schema definition may optionally use a `Set` function of type `SchemaSetFunc`. It may be
 used when a default hashing function (which calcluates hash based on all fields) is not suitable.
 
+## Filtering
+
+Filters provide the ability of retrieving a data source by criteria other than the plain name. They depend on
+the query engine in go-vcloud-director. Only types supported by such engine can have a filter.
+
+To add filtering for a new data source:
+
+1. make sure the underlying entity supports filtering (See "Supporting a new type in the query engine" in 
```

Comment:
Thanks for this section! Nice how you link the govcd coding guidelines as well. Few nits below.

```suggestion
1. Make sure the underlying entity supports filtering (See "Supporting a new type in the query engine" in 
```

Reply:
fixed